client/src/modules/payroll/weekend_configuration/modals/config.modal.js

```diff
@@ -0,0 +1,115 @@
+angular.module('bhima.controllers')
+  .controller('WeekEndConfigModalController', WeekEndConfigModalController);
+
+WeekEndConfigModalController.$inject = [
+  '$state', 'ConfigurationWeekEndService', 'NotifyService', 'appcache',
+];
+
+function WeekEndConfigModalController($state, Config, Notify, AppCache) {
+  var vm = this;
+  vm.weekend = {};
+
+  var cache = AppCache('RubricModal');
+
+  if ($state.params.creating || $state.params.id) {
+    vm.stateParams = cache.stateParams = $state.params;
+  } else {
+    vm.stateParams = cache.stateParams;
+  }
+  vm.isCreating = vm.stateParams.creating;
+
+  vm.weekDays = [
```

Comment:
This looks like it should go in a service - you might want to use it again somewhere else.